src/pages/Compare/index.spec.tsx

```diff
@@ -836,7 +836,9 @@ describe(__filename, () => {
     if (!diff) {
       throw new Error('diff was unexpectedly empty');
     }
-    const map = new ForwardComparisonMap(diff);
+    const getterFromFactory = getCodeLineAnchorGetter({
+      compareInfo: { diff, mimeType: 'test' },
```

Comment:
This looks fragile. I think you can just get a `compareInfo` object using the same mapper:

```diff
diff --git a/src/pages/Compare/index.spec.tsx b/src/pages/Compare/index.spec.tsx
index de2254f..f6b4ee9 100644
--- a/src/pages/Compare/index.spec.tsx
+++ b/src/pages/Compare/index.spec.tsx
@@ -812,11 +812,13 @@ describe(__filename, () => {
   });
 
   it('configures VersionFileViewer with a file', () => {
+    const addonId = 1;
     const baseVersionId = 1;
     const version = { ...fakeVersionWithDiff, id: baseVersionId + 1 };
     const headVersionId = version.id;
 
-    const { root } = loadDiffAndRender({
+    const { store, root } = loadDiffAndRender({
+      addonId,
       baseVersionId,
       headVersionId,
       version,
@@ -832,12 +834,13 @@ describe(__filename, () => {
       }),
     );
 
-    const diff = createInternalDiff({ baseVersionId, headVersionId, version });
-    if (!diff) {
-      throw new Error('diff was unexpectedly empty');
-    }
     const getterFromFactory = getCodeLineAnchorGetter({
-      compareInfo: { diff, mimeType: 'test' },
+      compareInfo: getCompareInfo(
+        store.getState().versions,
+        addonId,
+        baseVersionId,
+        headVersionId,
+      ),
     });
 
     const getCodeLineAnchor = viewer.prop('getCodeLineAnchor');
```

Reply:
Nice, thanks!